src/cryptonote_basic/cryptonote_basic.h

```diff
@@ -346,79 +348,101 @@ namespace cryptonote
 
   inline transaction::transaction(const transaction &t):
     transaction_prefix(t),
-    hash_valid(false),
-    prunable_hash_valid(false),
-    blob_size_valid(false),
+    hash_valid(t.is_hash_valid()),
+    prunable_hash_valid(t.is_prunable_hash_valid()),
+    blob_size_valid(t.is_blob_size_valid()),
     signatures(t.signatures),
     rct_signatures(t.rct_signatures),
+    hash(t.hash),
+    prunable_hash(t.prunable_hash),
+    blob_size(t.blob_size),
     pruned(t.pruned),
     unprunable_size(t.unprunable_size.load()),
     prefix_size(t.prefix_size.load())
+  {}
+
+  inline transaction::transaction(transaction &&t):
+    transaction_prefix(std::move(t)),
+    hash_valid(t.is_hash_valid()),
+    prunable_hash_valid(t.is_prunable_hash_valid()),
+    blob_size_valid(t.is_blob_size_valid()),
+    signatures(std::move(t.signatures)),
+    rct_signatures(std::move(t.rct_signatures)),
+    hash(std::move(t.hash)),
+    prunable_hash(std::move(t.prunable_hash)),
+    blob_size(std::move(t.blob_size)),
+    pruned(std::move(t.pruned)),
+    unprunable_size(t.unprunable_size.load()),
+    prefix_size(t.prefix_size.load())
   {
-    if (t.is_hash_valid())
-    {
-      hash = t.hash;
-      set_hash_valid(true);
-    }
-    if (t.is_blob_size_valid())
-    {
-      blob_size = t.blob_size;
-      set_blob_size_valid(true);
-    }
-    if (t.is_prunable_hash_valid())
-    {
-      prunable_hash = t.prunable_hash;
-      set_prunable_hash_valid(true);
-    }
+    t.set_null();
   }
 
   inline transaction &transaction::operator=(const transaction &t)
   {
+    if (this == std::addressof(t))
```

Comment:
Curious why you prefer addressof instead of `&`?

Reply:
I think the recommendation is to primarily use in templated code, but I've been using it for any type in case an `operator&` is ever defined. Its probably a pessimistic, but I could switch to `&` here if desired.

Reply:
Keep it, that makes sense. Was moreso curiousity on my end.